Model/Order/GetOrderFromQuoteId.php

```diff
@@ -0,0 +1,52 @@
+<?php
+declare(strict_types=1);
+
+namespace Bold\Checkout\Model\Order;
+
+use Bold\Checkout\Api\Order\GetOrderFromQuoteIdInterface;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Sales\Api\Data\OrderInterface;
+use Magento\Sales\Api\OrderRepositoryInterface;
+
+class GetOrderFromQuoteId implements GetOrderFromQuoteIdInterface
+{
+    /**
+     * @var OrderRepositoryInterface
+     */
+    private $orderRepository;
+
+    /**
+     * @var SearchCriteriaBuilder
+     */
+    private $searchCriteriaBuilder;
+
+    public function __construct(
+        OrderRepositoryInterface $orderRepository,
+        SearchCriteriaBuilder $searchCriteriaBuilder
+    ) {
+        $this->orderRepository = $orderRepository;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+    }
+
+    /**
+     * @param int $quoteId
+     * @return OrderInterface
+     * @throws LocalizedException
+     */
+    public function getOrder(int $quoteId): OrderInterface
+    {
+        $searchCriteria =  $this->searchCriteriaBuilder
```

Comment:
We may want to confirm with a Magento expert if quote id's are long lived. I thought I had heard something in the past that quote id's expire or can be cleared out by the merchant. If so, this may result in this endpoint being a new failure point for Bold.

Reply:
Calin is correct, to the best of my knowledge.

Reply:
With the change in approach for the ticket this was related to it looks like this endpoint won't be needed. That on top of it sounding like this isn't a stable approach anyway I'll be closing this once the PR in the other project has been merged. Thank you both!